### ISSUE TITLE
graphql api returns 500 http codes on unknown types

### ISSUE THREAD

**Issue by massimilianobraglia** [thecodingmachine/graphqlite#336]
Hey there,

```graphql
query foobar($foobar: Foobar!){
  foobar(foobar: $foobar){
    foobar
  }
}
```

Calling this query on an environment that has no `Foobar` type defined results in a 500 http response like in this screenshot.

Is it normal? Shouldn't it be a 400 response?

![image](https://user-images.githubusercontent.com/17158942/114885983-7633d480-9e07-11eb-9486-f8ac33a89723.png)

**Comment by oojacoboo** [comment]
Yes.  I think there are a few cases where the http response codes aren't being properly set.  Care to submit a pull request?

**Comment by massimilianobraglia** [comment]
I'll try to find some time for this one! Not sure when :/

**Comment by oojacoboo** [comment]
@massimilianobraglia be a great time to get a PR merged in.  Thanks.

**Comment by sebcar** [comment]
I'll try to solve this issue. @oojacoboo any suggestions about the test case?

**Comment by oojacoboo** [comment]
@sebcar I think an example like the one mentioned should be sufficient here.  There are a few other cases as well, where GQL syntax results in a 500.  I'm not sure of all these cases though.  GraphQLite uses Webonyx's GQL parser under the hood.  I don't think we'll be able to wrap that in a try/catch either, unless there is a specific Exception being thrown for these parsing errors.

**Comment by sebcar** [comment]
I can't find a way in the test case to get the execution result for CannotMapTypeException

**Comment by oojacoboo** [comment]
@sebcar If there is an exception being thrown, why would there be a result?  Presumably, execution was halted before there was a result.

**Comment by sebcar** [comment]
@oojacoboo that is true. I was trying to use the HttpCodeDeciderTest
